src/overlays/actors/ovl_Demo_Gj/z_demo_gj.c

```diff
@@ -96,241 +85,1396 @@ static ColliderCylinderInitType1 sCylinderInit = {
+    DemoGj_DrawCommon(this, globalCtx, &D_06002160);
+}
+
+void func_8097B8E8(DemoGj* this, GlobalContext* globalCtx) {
+    DemoGj_InitSetIndexes(this, globalCtx, 17, 0, NULL);
+    DemoGj_Cylinder_Init(this, globalCtx, &this->cylinders[0], &sCylinderInit3);
+}
+
+void func_8097B930(DemoGj* this, GlobalContext* globalCtx) {
+}
+
+void func_8097B93C(DemoGj* this, GlobalContext* globalCtx, Vec3f* arg2) {
+    Vec3f pos;
+
+    pos.x = this->dyna.actor.world.pos.x;
+    pos.y = this->dyna.actor.world.pos.y;
+    pos.z = this->dyna.actor.world.pos.z;
+    func_80978C20(this, globalCtx, &pos, arg2);
+
+    pos.x = this->dyna.actor.world.pos.x;
+    pos.y = this->dyna.actor.world.pos.y + 100.0f;
+    pos.z = this->dyna.actor.world.pos.z;
+    func_80978C20(this, globalCtx, &pos, arg2);
+}
+
+void func_8097B9BC(DemoGj* this, GlobalContext* globalCtx) {
+    Vec3f* scale = &this->dyna.actor.scale;
+
+    if (func_8097983C(this, globalCtx)) {
+        DemoGj_InitCommon(this, globalCtx, &D_06003AF0);
+        this->updateIndex = 20;
+        this->drawIndex = 18;
+        scale->x *= 0.8f;
+        scale->y *= 0.8f;
+        scale->z *= 0.8f;
+    }
+}
+
+void func_8097BA48(DemoGj* this, GlobalContext* globalCtx) {
+    Actor* actor = &this->dyna.actor;
+    ColliderCylinder* cylinder = &this->cylinders[0];
+    Collider* collider = &cylinder->base;
+    CollisionCheckContext* colChkCtx = &globalCtx->colChkCtx;
+
+    Vec3f vec1;
+    Vec3f vec2;
+
+    if (func_809797E4(this, 4)) {
+        Actor_Kill(actor);
+    } else if (DemoGj_Cylinder_HasExploded(this, globalCtx, cylinder)) {
+        vec1 = D_8097BEC4;
+
+        DemoGj_DropCollectible(this, globalCtx);
+        func_8097B93C(this, globalCtx, &vec1);
+
+        Actor_Kill(actor);
+    } else if (this->flag3) {
+        vec2 = this->unk_26C;
+        vec2.y = 0.0f;
+
+        DemoGj_DropCollectible(this, globalCtx);
+        func_8097B93C(this, globalCtx, &vec2);
+
+        Actor_Kill(actor);
+    }
+
+    Collider_UpdateCylinder(actor, cylinder);
+    CollisionCheck_SetAC(globalCtx, colChkCtx, collider);
+}
+
+void DemoGj_Update17(DemoGj* this, GlobalContext* globalCtx) {
+    DemoGj_FindGanon(this, globalCtx);
+    func_8097B9BC(this, globalCtx);
+}
+
+void DemoGj_Update20(DemoGj* this, GlobalContext* globalCtx) {
+    func_8097BA48(this, globalCtx);
+    func_8097B930(this, globalCtx);
+}
+
+void DemoGj_Draw18(DemoGj* this, GlobalContext* globalCtx) {
+    DemoGj_DrawCommon(this, globalCtx, &D_06003710);
+}
+
+void DemoGj_Update(Actor* thisx, GlobalContext* globalCtx) {
+    DemoGj* this = THIS;
+    DemoGjActionFunc func;
+
+    if (this->updateIndex < 0 || this->updateIndex >= 21 || (func = sUpdateFuncs[this->updateIndex]) == NULL) {
+        // The main mode is abnormal!!!!!!!!!!!!!!!!!!!!!!!!!
+        osSyncPrintf(VT_FGCOL(RED) "メインモードがおかしい!!!!!!!!!!!!!!!!!!!!!!!!!\n" VT_RST);
+        return;
+    }
+
+    func(this, globalCtx);
```

Comment:
```suggestion

    if (this->updateIndex < 0 || this->updateIndex >= 21 || sUpdateFuncs[this->updateIndex] == NULL) {
        // The main mode is abnormal!!!!!!!!!!!!!!!!!!!!!!!!!
        osSyncPrintf(VT_FGCOL(RED) "メインモードがおかしい!!!!!!!!!!!!!!!!!!!!!!!!!\n" VT_RST);
        return;
    }

    sUpdateFuncs[this->updateIndex](this, globalCtx);
```
You don't need the temp